website/docs/advanced/search_path.md

```diff
@@ -3,15 +3,94 @@ id: search_path
 title: Config Search Path
 ---
 
-The Config Search Path is a list of paths that are searched in order to find configs. It is similar to
-the Python PYTHONPATH.
+import {ExampleGithubLink} from "@site/src/components/GithubLink"
 
+The Config Search Path is a list of paths that Hydra searches in order to find configs. It is similar to
+the Python `PYTHONPATH`. 
  - When a config is requested, The first matching config in the search path is used.
- - Each search path element has a schema prefix such as file:// or pkg:// that is corresponding to a ConfigSourcePlugin.
- - `SearchPathPlugin` can manipulate the search path.
+ - Each search path element has a schema prefix such as `file://` or `pkg://` that is corresponding to a `ConfigSourcePlugin`.
 
 You can inspect the search path and the configurations loaded by Hydra by turning on verbose logging for the `hydra` logger:
 
-```text
+```bash
 $ python my_app.py hydra.verbose=hydra
-```
\ No newline at end of file
+```
+
+There's a few options to modify the config search path so Hydra could discover 
+the configurations. You can pick and choose some or all of the following methods below
+to provide configuration to your apps.
+
+### Set `config_name`/`config_path` in`hydra.main()`
+This is the most common use case: all config files are grouped under one directory and
+has a relative path to the application. You can add them to the search path by setting `config_path` 
+or `config_name`.
+
+### Override `hydra.searchpath` config
+
+<ExampleGithubLink text="Example application" to="examples/advanced/config_search_path"/>
+
+Sometimes your configurations are not grouped nicely in one folder - 
+you may need to read configurations from multiple locations, so passing one directory to `config_path` 
+won't work. In this case, you can pass in a list of config paths via `hydra.searchpath` in 
+the primary config or override from the command line.
+
+Let's say we have the following configuration and application structure:
+```bash
+$ tree  config_search_path
+config_search_path
+├── __init__.py
+├── conf
+│   └── config.yaml
+├── dataset
+│   ├── cifar10.yaml
+│   └── imagenet.yaml
+└── my_app.py
+```
+where `conf/config.yaml` is the primary config for `my_app.py`. However at the same time, we also 
+want to reference configuration from `dataset` in `config.yaml`. In this case, we can add `dataset` to 
+`hydra.searchpath` so Hydra could discover the configurations under `dataset`.
```

Comment:
You have added a config searthpath inside another config searcpath. this is very bad example for people to follow :).